Viewer Plot window is broken

With version 0.18.0, the viewer plot window is broken.
This was seen by both Dominik and myself.

![issue_fact_tools](https://cloud.githubusercontent.com/assets/17047733/22148302/92447a3e-df0d-11e6-8131-6075f1c02b0e.png)

```
java -jar target/fact-tools-0.18.0.jar examples/viewer.xml 
Jan 20 2017 12:35:46 [ERROR] stream.io.SourceURL: Failed to register handler for protocol 'hdfs'
Jan 20 2017 12:35:46 [INFO ] stream.util.XIncluder:    url expanded to: 'classpath:/default/data/calibration.xml'
Jan 20 2017 12:35:46 [INFO ] stream.util.XIncluder:    url expanded to: 'classpath:/default/data/extraction.xml'
Jan 20 2017 12:35:46 [INFO ] stream.util.XIncluder:    url expanded to: 'classpath:/default/data/cleaning.xml'
Jan 20 2017 12:35:46 [INFO ] stream.util.XIncluder:    url expanded to: 'classpath:/default/data/pedestalParameters.xml'
Jan 20 2017 12:35:47 [INFO ] stream.runtime.ProcessContainer: ######## Sources ########
Jan 20 2017 12:35:47 [INFO ] stream.runtime.ProcessContainer: ########fact.io.hdureader.FITSStream@470e2030########
Jan 20 2017 12:35:47 [INFO ] stream.runtime.ProcessContainer: 	==> DefaultProcess['2']
Jan 20 2017 12:35:47 [INFO ] stream.runtime.ProcessContainer: ######## RootSources ########
Jan 20 2017 12:35:47 [INFO ] stream.runtime.ProcessContainer: fact.io.hdureader.FITSStream@470e2030
Jan 20 2017 12:35:47 [INFO ] stream.runtime.ProcessContainer: ######## NonRefSinks ########
Jan 20 2017 12:35:47 [INFO ] stream.runtime.ProcessContainer: ProcessContainer is initialized and ready to start:stream.runtime.ProcessContainer@1554909b
Jan 20 2017 12:35:47 [INFO ] stream.run: Starting process-container...
Jan 20 2017 12:35:47 [INFO ] stream.runtime.ProcessContainer: Initializing life-cycle objects...
Jan 20 2017 12:35:47 [INFO ] stream.runtime.ProcessContainer: Creating 1 active processes...
Jan 20 2017 12:35:47 [INFO ] fact.ShowViewer: Opening viewer on OS: Linux
Jan 20 2017 12:35:49 [INFO ] org.reflections.Reflections: Reflections took 80 ms to scan 1 urls, producing 56 keys and 288 values 
Jan 20, 2017 12:35:59 PM com.google.common.eventbus.EventBus$LoggingSubscriberExceptionHandler handleException
SEVERE: Could not dispatch event: fact.hexmap.ui.components.selectors.IntervallMarkerKeySelector[,0,0,0x0,invalid,layout=java.awt.BorderLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=9,maximumSize=,minimumSize=,preferredSize=] to public void fact.hexmap.ui.components.selectors.KeySelector.handleEventChange(org.apache.commons.math3.util.Pair)
Jan 20, 2017 12:35:59 PM com.google.common.eventbus.EventBus$LoggingSubscriberExceptionHandler handleException
SEVERE: Could not dispatch event: fact.hexmap.ui.components.selectors.TimeSeriesKeySelector[,0,0,0x0,invalid,layout=java.awt.BorderLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=9,maximumSize=,minimumSize=,preferredSize=] to public void fact.hexmap.ui.components.selectors.KeySelector.handleEventChange(org.apache.commons.math3.util.Pair)
```

--- a/src/main/java/fact/Utils.java
+++ b/src/main/java/fact/Utils.java
@@ -20,9 +20,9 @@ import java.util.List;
 
 import org.apache.commons.lang3.ArrayUtils;
 /**
- * 
+ *
  * @author Kai Bruegge &lt;kai.bruegge@tu-dortmund.de&gt;
- * 
+ *
  */
 public class Utils {
 	static Logger log = LoggerFactory.getLogger(Utils.class);
@@ -70,7 +70,7 @@ public class Utils {
 	 * This takes a data array (of length pixels * roi) and returns an
 	 * array(length = roi) in which each entry is the average over all the
 	 * values of the other pixels in that slice.
-	 * 
+	 *
 	 * @param data
 	 *            array of length pixels*region of interest
 	 * @param roi
@@ -100,7 +100,7 @@ public class Utils {
 	 * list of lists. Each list containing one separate set. Does a BFs search.
 	 * See the wikipedia article on BFS. This version is not as memory efficient
 	 * as it could be.
-	 * 
+	 *
 	 * @param showerPixel
 	 *            the list to search in
 	 * @return A list of lists.
@@ -214,6 +214,9 @@ public class Utils {
 	 * @return can be null!
 	 */
 	public static double[] toDoubleArray(Serializable arr) {
+		if (arr == null) {
+			return null;
+		}
 		if (arr.getClass().isArray()) {
 			Class<?> clazz = arr.getClass().getComponentType();
 			if (clazz.equals(float.class)) {
@@ -262,7 +265,7 @@ public class Utils {
 	 * data item. If one of the keys is not in the item a RuntimeException will
 	 * be thrown containing a message detailing which processor is causing the
 	 * error
-	 * 
+	 *
 	 * @param item
 	 * @param keys
 	 */
@@ -296,7 +299,7 @@ public class Utils {
 	 * This method tries to find the key in the data item and tries to cast them
 	 * into the type given by the type parameter. If it fails it will throw a
 	 * RuntimeException with a message containing information about the error.
-	 * 
+	 *
 	 * @param item
 	 * @param key
 	 * @param type
@@ -331,7 +334,7 @@ public class Utils {
 	 * ellipse coordinates (l, t). The ellipse coordinate system is defined by
 	 * the center of gravity (x,y) and the angle between the major axis and the
 	 * camera x-axis (delta in radians).
-	 * 
+	 *
 	 * @param x
 	 * @param y
 	 * @param cogX
@@ -356,20 +359,20 @@ public class Utils {
 
 		return c;
 	}
-    
+
     public static double calculateDistancePointToShowerAxis(double cogx, double cogy, double delta, double x, double y){
     	double distance = 0;
-    	
+
     	double r0 = (x-cogx)*Math.cos(delta)+(y-cogy)*Math.sin(delta);
-    	
+
     	distance = Math.sqrt( Math.pow(cogx-x+r0*Math.cos(delta),2)+Math.pow(cogy-y+r0*Math.sin(delta),2) );
-    	
+
     	return distance;
     }
 
 	/**
 	 * Sum up array
-	 * 
+	 *
 	 * @param a
 	 * @return
 	 */
@@ -390,7 +393,7 @@ public class Utils {
 
 	/**
 	 * Elementwise multiplication of arrays
-	 * 
+	 *
 	 * @param a
 	 * @param b
 	 * @return double array containing a[i] * b[i]
@@ -416,7 +419,7 @@ public class Utils {
 
 		return ret;
 	}
-	
+
 	public static int[] getValidWindow(int start, int size, int validLeft, int validRight)
 	{
 		if (size < 0)
@@ -428,7 +431,7 @@ public class Utils {
 			throw new RuntimeException("validLeft > validRight! validLeft: "+ validLeft + " validRight: " + validRight);
 		}
 		int[] window = {start,start+size};
-		
+
 		if (start > validRight)
 		{
 			window[0] = validRight;
@@ -451,10 +454,10 @@ public class Utils {
 		}
 		return window;
 	}
-	
-	
+
+
 	public static void checkWindow(int start, int size, int validLeft, int validRight)
-	{	
+	{
 		if (size < 0)
 		{
 			throw new RuntimeException("Size for window < 0! size: "+ size);
@@ -468,7 +471,7 @@ public class Utils {
 		{
 			String message = "start + size > validRight. start+size/validRight: " + (start+size) + "/" + validRight;
 			throw new RuntimeException(message);
-		}	
+		}
 	}
 
 

--- a/src/main/java/fact/hexmap/ui/components/selectors/IntervallMarkerKeySelector.java
+++ b/src/main/java/fact/hexmap/ui/components/selectors/IntervallMarkerKeySelector.java
@@ -24,15 +24,12 @@ public class IntervallMarkerKeySelector extends KeySelector {
         for (String key : item.keySet()) {
             try {
                 IntervalMarker[] i = (IntervalMarker[]) item.get(key);
-                if(i.length == 1440) {
+                if(i != null && i.length == 1440) {
                     newItems.add(new SeriesKeySelectorItem(key, Color.LIGHT_GRAY, this));
                 }
             } catch (ClassCastException e){
                 continue;
-            } catch (RuntimeException e) {
-                continue;
             }
-
         }
         return newItems;
     }

--- a/src/main/java/fact/hexmap/ui/components/selectors/IntervallMarkerKeySelector.java
+++ b/src/main/java/fact/hexmap/ui/components/selectors/IntervallMarkerKeySelector.java
@@ -29,6 +29,8 @@ public class IntervallMarkerKeySelector extends KeySelector {
                 }
             } catch (ClassCastException e){
                 continue;
+            } catch (RuntimeException e) {
+                continue;
             }
 
         }

--- a/src/main/java/fact/hexmap/ui/components/selectors/TimeSeriesKeySelector.java
+++ b/src/main/java/fact/hexmap/ui/components/selectors/TimeSeriesKeySelector.java
@@ -6,6 +6,7 @@ import fact.hexmap.ui.events.TimeSeriesSelectionChangedEvent;
 import stream.Data;
 
 import java.awt.*;
+import java.io.Serializable;
 import java.util.HashSet;
 import java.util.Set;
 
@@ -27,14 +28,15 @@ public class TimeSeriesKeySelector extends KeySelector {
     public Set<SeriesKeySelectorItem> filterItems(Data item) {
         Set<SeriesKeySelectorItem> newItems = new HashSet<>();
         for (String key : item.keySet()) {
-            try {
-                double[] series = Utils.toDoubleArray(item.get(key));
-                if (series != null && (series.length > 1440) && (series.length % 1440 == 0)) {
-                    SeriesKeySelectorItem newSelectorItem = new SeriesKeySelectorItem(key, Color.GRAY, this);
-                    newItems.add(newSelectorItem);
-                }
-            } catch(RuntimeException e) {
+
+            Serializable value = item.get(key);
+            if (value == null){
                 continue;
+            }
+            double[] series = Utils.toDoubleArray(value);
+            if (series != null && (series.length > 1440) && (series.length % 1440 == 0)) {
+                SeriesKeySelectorItem newSelectorItem = new SeriesKeySelectorItem(key, Color.GRAY, this);
+                newItems.add(newSelectorItem);
             }
         }
         return newItems;

--- a/src/main/java/fact/hexmap/ui/components/selectors/TimeSeriesKeySelector.java
+++ b/src/main/java/fact/hexmap/ui/components/selectors/TimeSeriesKeySelector.java
@@ -16,7 +16,6 @@ import java.util.Set;
  */
 public class TimeSeriesKeySelector extends KeySelector {
 
-
     @Override
     public void selectionUpdate() {
 
@@ -28,12 +27,14 @@ public class TimeSeriesKeySelector extends KeySelector {
     public Set<SeriesKeySelectorItem> filterItems(Data item) {
         Set<SeriesKeySelectorItem> newItems = new HashSet<>();
         for (String key : item.keySet()) {
-            double[] series = Utils.toDoubleArray(item.get(key));
-//                double[] series = (double[]) item.get(key);
-            if (series !=  null &&(series.length > 1440) && (series.length % 1440 == 0)){
-                SeriesKeySelectorItem newSelectorItem = new SeriesKeySelectorItem(key, Color.GRAY, this);
-                //newSelectorItem.setSelector(this);
-                newItems.add(newSelectorItem);
+            try {
+                double[] series = Utils.toDoubleArray(item.get(key));
+                if (series != null && (series.length > 1440) && (series.length % 1440 == 0)) {
+                    SeriesKeySelectorItem newSelectorItem = new SeriesKeySelectorItem(key, Color.GRAY, this);
+                    newItems.add(newSelectorItem);
+                }
+            } catch(RuntimeException e) {
+                continue;
             }
         }
         return newItems;

